### PR TITLE
ENH: Added `CARDReadsKmerAnalysis` and `CARDMAGsKmerAnalysis` types

### LINKED ISSUE #47: ENH: Add new sematic types `CARDReadsKmerAnalysis` and `CARDMAGsKmerAnalysis`
For #21 two new actions will be added that run `kmer-query` of RGI for the reads `CARDAlleleAnnotation` and mags `CARDAnnotation` output.
The outputs of these two new functions need two new types called `CARDReadsKmerAnalysis` and `CARDMAGsKmerAnalysis`.

### PR DESCRIPTION
**Pull request by VinzentRisch**
This PR closes #47

- Added `CARDReadsGeneKmerAnalysis`, `CARDReadsAlleleKmerAnalysisand` and `CARDMAGsKmerAnalysis` types.
- The two actions that will be added described in #21 will output these types.

### Set up an environment
```bash
CONDA_SUBDIR=osx-64 mamba create -yn q2-amr \
  -c conda-forge -c bioconda -c qiime2 -c defaults \
  -c https://packages.qiime2.org/qiime2/2024.2/shotgun/released/ \
  qiime2 q2cli q2templates q2-types rgi

conda activate q2-amr
conda config --env --set subdir osx-64

pip install --no-deps --force-reinstall \
  git+https://github.com/misialq/rgi.git@py38-fix \
```

### Run it locally 
1. First, clone the repo and checkout the PR branch:
```bash
git clone https://github.com/bokulich-lab/q2-amr.git
cd q2-amr
git fetch origin pull/48/head:pr-48
git checkout pr-48
pip install -e .
```
2. Download test files 
[PR_48.zip](https://github.com/bokulich-lab/q2-amr/files/14435526/PR_48.zip)

3. Test it out!
Run these commands in the PR_48 directory
```bash
qiime tools import --input-path data_reads_allele --output-path reads_allele_kmer.qza --type "SampleData[CARDReadsAlleleKmerAnalysis]"
qiime tools import --input-path data_reads_gene --output-path reads_gene_kmer.qza --type "SampleData[CARDReadsGeneKmerAnalysis]"
qiime tools import --input-path data_mags --output-path mags_kmer.qza --type "SampleData[CARDMAGsKmerAnalysis]"
 ```

### CODE REVIEW

**Comment by VinzentRisch** [comment]
Hi @ChristosMatzoros 
Can you look this over? its really small just a couple of types.

File: q2_amr/types/_format.py [hunk 1]
```diff
@@ -430,3 +430,215 @@ class CARDGeneAnnotationDirectoryFormat(MultiDirValidationMixin, model.Directory
...
+            "#_of_AMR_kmers",
+            "taxonomic_info",
+            "genomic_info",
+        ]
+        with open(str(self)) as json_file:
+            dict = json.load(json_file)
+
+        keys_obs = list(next(iter(dict.values())).keys())
+
+        if keys_obs != keys_exp:
```
Comment (ChristosMatzoros) [inline q2_amr/types/_format.py:480]: If you don't want a specific order to the columns, consider using a set comparison to ensure that all expected keys are present without enforcing the order. For example, use `set(keys_obs) != set(keys_exp)`

File: q2_amr/types/_format.py [hunk 2]
```diff
@@ -430,3 +430,215 @@ class CARDGeneAnnotationDirectoryFormat(MultiDirValidationMixin, model.Directory
...
+            "MAPQ",
+            "taxonomic_info",
+            "genomic_info",
+        ]
+        with open(str(self)) as json_file:
+            dict = json.load(json_file)
+
+        keys_obs = list(next(iter(dict.values())).keys())
+
+        if keys_obs != keys_exp:
```
Comment (ChristosMatzoros) [inline q2_amr/types/_format.py:600]: Same as my previous comment, change to comparing sets if you don't care about enforcing order to the columns.

**Review by ChristosMatzoros — CHANGES_REQUESTED** [review]

**Review by ChristosMatzoros — APPROVED** [review]
Hey @VinzentRisch , everything works as expected. You can merge!

**Pull request merged**